db/src/tx.rs

```diff
@@ -710,7 +714,7 @@ impl<'conn, 'a, W> Tx<'conn, 'a, W> where W: TransactWatcher {
         }
 
         db::update_partition_map(self.store, &self.partition_map)?;
-        self.watcher.done(self.schema)?;
+        self.watcher.done(&self.tx_id, self.schema)?;
```

Comment:
I would be inclined to give the watcher the tx _first_, not (or as well as) here. We know it at the top of the function. Consider a watcher that wants to log the full five-tuple of each datom… it can't begin its work until it gets here.